src/handlers.rs

```diff
@@ -312,27 +312,54 @@ fn handle_fetch_req(d: *const DenoC, base: &msg::Base) -> Box<Op> {
   let url = url.parse::<hyper::Uri>().unwrap();
   let client = Client::new();
 
-  let future = client.get(url).and_then(|res| {
+  let future = client.get(url).and_then(move |res| {
     let status = res.status().as_u16() as i32;
+
+    let headers = {
+      let map = res.headers();
+      let keys = map
+        .keys()
+        .map(|s| s.as_str().to_string())
```

Comment:
This is fine for now, but according to the hyper documentation 
```
In practice, HTTP header field values are usually valid ASCII. However, the HTTP spec allows for a header value to contain opaque bytes as well. In this case, the header field value is not able to be represented as a string.
```
Which is totally annoying. I don't think I've ever encountered non-string header values. But if someone comes up with a counter example we might have to modify this.

Reply:
@ry It might be beneficial to know that It's talking about this part of the spec [Section 3.2.4 [RFC7230] ](https://tools.ietf.org/html/rfc7230#section-3.2)